rpc/ethereum/types/utils.go

```diff
@@ -256,25 +256,77 @@ func BaseFeeFromEvents(events []abci.Event) *big.Int {
 	return nil
 }
 
-// TxIndexFromEvents parses the tx index from cosmos events
-func TxIndexFromEvents(events []abci.Event) (uint64, error) {
+// FindTxAttributes returns the msg index of the eth tx in cosmos tx, and the attributes,
+// returns -1 and nil if not found.
+func FindTxAttributes(events []abci.Event, txHash string) (int, map[string]string) {
+	msgIndex := -1
 	for _, event := range events {
 		if event.Type != evmtypes.EventTypeEthereumTx {
 			continue
 		}
 
-		for _, attr := range event.Attributes {
-			if bytes.Equal(attr.Key, []byte(evmtypes.AttributeKeyTxIndex)) {
-				result, err := strconv.ParseInt(string(attr.Value), 10, 64)
-				if err != nil {
-					return 0, err
-				}
-				if result < 0 {
-					return 0, errors.New("negative tx index")
-				}
-				return uint64(result), nil
+		msgIndex++
+		value := FindAttribute(event.Attributes, []byte(evmtypes.AttributeKeyEthereumTxHash))
+		if bytes.Equal(value, []byte(txHash)) {
+			// convert attributes to map for later lookup
+			attrs := make(map[string]string, len(event.Attributes))
+			for _, attr := range event.Attributes {
+				attrs[string(attr.Key)] = string(attr.Value)
 			}
+			return msgIndex, attrs
+		}
```

Comment:
use `if !bytes.Equal -> continue` to improve readability 👍

Reply:
done